### PR TITLE
Add Mediapipe Tensors support

### PR DESCRIPTION
**Pull request by atobiszei**
* Additionally fix sporadic in model manager test by removing unnecessary assertion
* Minor tweaks in MP execution logs
* TfLite Tensors support in (de)serialization is turned off for 2023.1

Need to raise threshold due to cppclean complaining that
we need to have static specifier in template specialization which is not
possible (this specifier can be added only in template declaration).

CVS-116277

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -89,7 +89,7 @@ http_archive(
 git_repository(
     name = "mediapipe",
     remote = "https://github.com/openvinotoolkit/mediapipe",
-    commit = "ac9ff003cd735f96a2e5501e32b2c941f6d4543d", # Updated CAPI calls (#21)
+    commit = "18eddf82f473ebafa3c017805d3049d29b5404f6", # Updated CAPI calls (#21)
```
Comment (atobiszei) [inline WORKSPACE:92]: Commit will be bumped again after related MP fork repo MR will be merged to main

**Review by rasapala — APPROVED** [review]
LGTM

File: WORKSPACE [hunk 2]
```diff
@@ -213,6 +213,7 @@ http_archive(
         # Diff is generated with a script, don't update it manually.
         "@mediapipe//third_party:org_tensorflow_custom_ops.diff",
         "tf.patch",
+        "tf_graph_info_multilinecomment.patch",
```
Comment (dkalinowski) [inline WORKSPACE:216]: There is commit by @michalkulakowski added today that fixes cpp benchmark app building. It also uses TF. Michal needed to bring all patches from OVMS to the benchmark app. Can we confirm this commit does not break cpp benchmark app again?

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 3]
```diff
@@ -302,14 +462,56 @@ static Status createPacketAndPushIntoGraph(const std::string& name, const KFSReq
...
+    return StatusCode::OK;
+}
+
+static Status createPacketAndPushIntoGraphTfLiteTensor(const std::string& name, const KFSRequest& request, ::mediapipe::CalculatorGraph& graph, const std::unique_ptr<tflite::Interpreter>& interpreter, size_t tfLiteTensorIdx) {
+    if (name.empty()) {
+        SPDLOG_DEBUG("Creating Mediapipe graph inputs name failed for: {}", name);
+        return StatusCode::MEDIAPIPE_GRAPH_ADD_PACKET_INPUT_STREAM;
+    }
+    SPDLOG_DEBUG("Tensor to deserialize:\"{}\"", name);
+    if (request.raw_input_contents().size() == 0) {
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:495]: What about 0-dim requests?

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 4]
```diff
@@ -101,7 +214,48 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
...
+    }
+    auto inputIndex = requestInputItr - request.inputs().begin();
+    auto& bufferLocation = request.raw_input_contents().at(inputIndex);
+    try {
+        auto datatype = getPrecisionAsTfLiteDataType(KFSPrecisionToOvmsPrecision(requestInputItr->datatype()));
+        std::vector<int> rawShape;
+        for (int i = 0; i < requestInputItr->shape().size(); i++) {
+            if (requestInputItr->shape()[i] < 0) {
+                std::stringstream ss;
+                ss << "Negative or zero dimension size is not acceptable: " << tensorShapeToString(requestInputItr->shape()) << "; input name: " << requestedName;
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:231]: Wrong message, zero is allowed in line 229

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 5]
```diff
@@ -101,7 +214,48 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
...
+                const std::string details = ss.str();
+                SPDLOG_DEBUG("[servable name: {} version: {}] Invalid shape - {}", request.model_name(), request.model_version(), details);
+                return Status(StatusCode::INVALID_SHAPE, details);
+            }
+            rawShape.emplace_back(requestInputItr->shape()[i]);
+        }
+        interpreter->SetTensorParametersReadWrite(
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:238]: Needs testing if passing nullptr as buffer is supported (0-data tensors)

File: src/test/mediapipeflow_test.cpp [hunk 6]
```diff
@@ -140,6 +155,45 @@ TEST_F(MediapipeTFTest, DummyInfer) {
...
+    inputs_info_t inputsMeta{{"in", {DUMMY_MODEL_SHAPE, precision}}};
+    std::vector<float> requestData{13.5, 0., 0, 0., 0., 0., 0., 0, 3., 67.};
+    preparePredictRequest(request, inputsMeta, requestData);
+    request.mutable_model_name()->assign(modelName);
+    ASSERT_EQ(impl.ModelInfer(nullptr, &request, &response).error_code(), grpc::StatusCode::OK);
+    size_t dummysInTheGraph = 1;
+    checkDummyResponse("out", requestData, request, response, dummysInTheGraph, 1, modelName);
+}
+
+TEST_F(MediapipeTfLiteTensorTest, DummyInfer) {
```
Comment (dkalinowski) [inline src/test/mediapipeflow_test.cpp:176]: We should also check scalar and 0-data scenario, for example method for creating 0-data tensor differs from creating one with data in OpenVINO

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 7]
```diff
@@ -61,7 +71,112 @@ static Status getRequestInput(google::protobuf::internal::RepeatedPtrIterator<co
...
+    if (!status.ok()) {
+        return status;
+    }
+    auto inputIndex = requestInputItr - request.inputs().begin();
+    auto& bufferLocation = request.raw_input_contents().at(inputIndex);
+    try {
+        auto datatype = KFSPrecisionToMPPrecision(requestInputItr->datatype());
+        std::vector<int> rawShape;
+        for (int i = 0; i < requestInputItr->shape().size(); i++) {
+            if (requestInputItr->shape()[i] <= 0) {
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:157]: We should allow and test zeros in shape

File: src/mediapipe_internal/mediapipegraphexecutor.cpp [hunk 8]
```diff
@@ -537,12 +772,18 @@ Status MediapipeGraphExecutor::infer(const KFSRequest* request, KFSResponse* res
         if (!status.ok()) {
             return status;
         }
+        ++insertedStreamPackets;
+    }
+    if (this->inputNames.size() > insertedStreamPackets) {
+        SPDLOG_DEBUG("Not all input packets created. Expected: {}, Actual: {}. Aborting execution of mediapipe graph: {}",
```
Comment (dkalinowski) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:778]: 3x `{}`, but 2 parameters - will crash at some point. Do we have test executing this code?

File: src/test/c_api_stress_tests.cpp [hunk 9]
```diff
@@ -278,7 +278,7 @@ class StressPipelineCustomNodesWithPreallocatedBuffersCapiConfigChanges : public
     void checkInferResponse(OVMS_InferenceResponse* response) override {
         ASSERT_NE(response, nullptr);
         uint32_t outputCount = 42;
-        ASSERT_CAPI_STATUS_NULL(OVMS_InferenceResponseGetOutputCount(response, &outputCount));
+        ASSERT_CAPI_STATUS_NULL(OVMS_InferenceResponseOutputCount(response, &outputCount));
```
Comment (dkalinowski) [inline src/test/c_api_stress_tests.cpp:281]: This is probably already fixed on main - PR not rebased?

File: src/test/mediapipe/config_mp_passthrough.json [hunk 10]
```diff
@@ -0,0 +1,9 @@
+{
+    "model_config_list": [],
```
Comment (dkalinowski) [inline src/test/mediapipe/config_mp_passthrough.json:2]: This test will probably fail when I merge my commit.
My commit fixes the issue with existence of `subconfig.json` in tests which adds dummy model.
Here you want to use dummy model but do not load dummy model in line 2.
Keep it in mind before merging PR

File: src/test/mediapipeflow_test.cpp (hunk 6 above)
Comment (atobiszei) [reply src/test/mediapipeflow_test.cpp:176]: TfLite Support is enabled only in calculator - not in OVMS

File: src/test/mediapipe/config_mp_passthrough.json (hunk 10 above)
Comment (atobiszei) [reply src/test/mediapipe/config_mp_passthrough.json:2]: Here test uses subconfig.json

File: src/test/c_api_stress_tests.cpp (hunk 9 above)
Comment (atobiszei) [reply src/test/c_api_stress_tests.cpp:281]: Fix was cherry-picked on this branch as well

File: WORKSPACE (hunk 1 above)
Comment (atobiszei) [reply WORKSPACE:92]: As soon as I have approvals.

File: src/mediapipe_internal/mediapipegraphexecutor.cpp (hunk 4 above)
Comment (atobiszei) [inline src/mediapipe_internal/mediapipegraphexecutor.cpp:231]:
```suggestion
                ss << "Negative dimension size is not acceptable: " << tensorShapeToString(requestInputItr->shape()) << "; input name: " << requestedName;
```

File: src/mediapipe_internal/mediapipegraphexecutor.cpp (hunk 5 above)
Comment (atobiszei) [reply src/mediapipe_internal/mediapipegraphexecutor.cpp:238]: TfLite tensors in OVMS (de)serialization will not be supported in this release

Comment (dkalinowski) [reply src/mediapipe_internal/mediapipegraphexecutor.cpp:238]: So why do we need this code?

File: WORKSPACE (hunk 2 above)
Comment (atobiszei) [reply WORKSPACE:216]: Benchmark builds OK with this MR without additional changes.

File: src/mediapipe_internal/mediapipegraphexecutor.cpp (hunk 7 above)
Comment (dkalinowski) [reply src/mediapipe_internal/mediapipegraphexecutor.cpp:157]: Is it scheduled in next task, next release or?

File: src/mediapipe_internal/mediapipegraphexecutor.cpp (hunk 3 above)
Comment (dkalinowski) [reply src/mediapipe_internal/mediapipegraphexecutor.cpp:515]: Is it scheduled in next task, next release or?

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**